### ISSUE TITLE
Subsecond precision for datetimes/durations

### ISSUE THREAD

**Issue by toihr** [typst/typst#4286]
### Description

The datetime function only supports integers for seconds, but sometimes for accuracy one might want to display fractional seconds. 

### Use Case

This would be useful for giving very accurate scientific datetimes. Which are often enough very precise numbers in the millisecond region.

**Comment by Leedehai** [comment]
Hi, could you elaborate on the use cases?

Keep in mind that Typst is for typesetting and not for scientific computing, so maybe more use cases are helpful to help justify the r feature.

**Comment by lzm0** [comment]
I agree. Without milliseconds or even nanoseconds on the Datetime Type is sometimes a show-stopper. Sometimes I would like to calculate the difference between two timestamps more precisely.
